cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -737,16 +735,19 @@ private void deleteApp(final ApplicationId appId, ApplicationSpecification spec)
     } catch (Exception e) {
       LOG.warn("Failed to unregister usage of app: {}", appId, e);
     }
+
+    // make sure the program profile metadata is removed
+    profileMetadataPublisher.removeProfileMetadata(appId, getProgramsWithType(appId, ProgramType.WORKFLOW,
```

Comment:
I think it's better to have a separate method for an app getting deleted that takes the app spec as an argument. Otherwise you have this weird usage where in one place .removeProfileMetata(appId, workflows) is called and in another place, .removeProfileMetadata(appId, schedules) is called.

Reply:
Done